docs/REST API/rest-api.md

```diff
@@ -6936,12 +6938,74 @@ Deletes the session specified by ID. When the user associated with the session n
 
 ## Software
 
+- [Upload a software](#upload-a-software)
+- [Delete a software](#delete-a-software)
 - [List software titles](#list-software-titles)
 - [List software versions](#list-software-versions)
 - [Count software](#count-software)
 - [Get software title](#get-software-title)
 - [Get software version](#get-software-version)
 
+### Upload a software
+
+Upload a software to install on macOS and Windows hosts during enrollment.
+
+`POST /api/v1/fleet/software`
+
+#### Parameters
+
+| Name            | Type    | In   | Description                                      |
+| ----            | ------- | ---- | --------------------------------------------     |
+| software        | file    | form | **Required**. Installer file. PKG or DMG for macOS and MSI or EXE for Windows hosts.   |
+| team_id         | integer | form | The team ID. If specified, the software will only be available to hosts assigned to this team. If not specified, the software will only be available to hosts that are not assigned to any team (No team).  |
+
+#### Example
+
+`POST /api/v1/fleet/software`
+
+##### Request header
+
+```http
+Content-Length: 8500
+Content-Type: multipart/form-data; boundary=------------------------d8c247122f594ba0
+```
+
+##### Request body
+
+```http
+--------------------------d8c247122f594ba0
+Content-Disposition: form-data; name="team_id"
+1
+--------------------------d8c247122f594ba0
+Content-Disposition: form-data; name="software"; filename="ZoomInstallFull.pkg"
+Content-Type: application/octet-stream
+
+[binary content of PKG file]
+
+--------------------------d8c247122f594ba0
+```
```

Comment:
Is this what the request body looks like for adding a bootstrap package? I would just make sure we're being consistent.

Reply:
Only difference was that bootstrap package example had `<BINARY DATA>` instead of `[binary content of PKG file]`. Updated to match bootstrap package.